clients/cobol-lsp-vscode-extension/src/services/Middleware.ts

```diff
@@ -0,0 +1,42 @@
+import { CancellationToken } from "vscode";
+import { ConfigurationParams, ConfigurationRequest } from "vscode-languageclient";
+import { CopybooksDownloader } from "./CopybooksDownloader";
+import { CopybooksPathGenerator } from "./CopybooksPathGenerator";
+
+export class Middleware {
+    constructor(
+            private copybooksPathGenerator: CopybooksPathGenerator,
+            private copybookDownloader: CopybooksDownloader) {}
+
+    public async handleConfigurationRequest(
+            params: ConfigurationParams,
+            token: CancellationToken,
+            next: ConfigurationRequest.HandlerSignature) {
+
+        if (params.items.length === 1) {
+            const section = params.items[0].section.split(".");
+            if (section[0] === "broadcom-cobol-lsp") {
+                switch (section[1]) {
+                    case "cpy-manager":
+                        return this.handleCpyManagerRequest();
+                    case "copybook":
+                        const cobolFileName = section[2];
```

Comment:
why `cobolFileName` is part of `section` but not part of `scopeUri`?
(sorry, I know it's not your code)

Reply:
It's my responsibility also. I could move it inside `scopeUri`.